src/pages/docs/guides/chat/build-livestream.mdx

```diff
@@ -0,0 +1,354 @@
+
+<Code>
+```javascript
+import { ErrorCode } from '@ably/chat';
+
+const room = chatClient.rooms.get('my-livestream');
+
+room.messages.send({text: 'some horrible message'}).then((message) => {
+    console.log('Message sent:', message);
+}).catch((error) => {
+    if (error.code === ErrorCode.MessageRejectedByModeration) {
+        console.log('Message rejected by moderation:', error.message);
+        return;
+    }
+
+    console.error('Message failed to send:', error);
+});
+```
+</Code>
+
+## Occupancy: Give volume to your livestream
+
+Ably's [occupancy](/docs/chat/rooms/occupancy) feature provides you with realtime metrics of how many users are engaging with your livestream. The metrics are based on the number of connections to the Ably service and can be delivered to clients in realtime. These can be used to build excitement in your streams, showing users how many others are experiencing it with them.
+
+* **Within a stream:** Show the number of active users in realtime.
+* **Across streams:** Aggregate occupancy data for stream lists or dashboards.
+
+## Reactions and typing indicators
+
+Enhance engagement with [room reactions](/docs/chat/rooms/reactions) and [typing indicators](/docs/chat/rooms/typing). Celebrate the winning goal or protest the controversial decision by enabling users to express sentiment en-masse.
+
+* **Room reactions:** Broadcast sentiment (e.g., 👍, ❤️) to all users.
+* **Typing indicators:** Show when users are composing messages.
+
+Typing indicators are less common in large livestream chat, where a large number of users are typing at once. However, in more intimate experiences such as supporting 1:1 and group chats, they can be a great way to enhance engagement. So a key design decision is whether or not user engagement will benefit from them.
+
+<Code>
+```javascript
+const room = chatClient.rooms.get('my-livestream');
+
+// Subscribe to typing events
+room.typing.subscribe((event) => {
+  console.log('Currently typing clients', event.currentlyTyping);
+});
+
+// Each time a user does a keystroke, call the keystroke method.
+// This method is self-throttling, to prevent large numbers of messages being sent
+// and keep costs down
+await room.typing.keystroke();
+
+// You can let the typing indicator stop naturally, or explicitly stop it
+await room.typing.stop();
+```
+</Code>
+
+Room reactions are a great way to enhance engagement in a livestream and enable users to quickly express sentiment to key moments in the stream. They are not persisted, and are optimized for scale. Ably's platform is designed to support the spikey traffic of livestream chat, ensuring that you don't encounter any performance issues during pivotal moments.
```

Comment:
Are we not going to mention message reactions?

Reply:
Perhaps less useful in the large single-room case, but I've added this as it relates to 1:1 DMs and small side-groups that we see in some FE experiences